pkg/controller/cloud/node_controller.go

```diff
@@ -321,7 +321,7 @@ func (cnc *CloudNodeController) AddCloudNode(obj interface{}) {
 		}
 
 		if zones, ok := cnc.cloud.Zones(); ok {
-			zone, err := zones.GetZone()
+			zone, err := getZoneByProviderIDOrName(zones, curNode)
```

Comment:
Thanks for fixing this. This originally always returned the zone in which the cloud controller manager was running.